server/games/ladder_game.py

```diff
@@ -22,31 +23,14 @@ def __init__(self, id, *args, **kwargs):
             await self.persist_rating_change_stats(new_ratings, rating='ladder')
 
     def is_winner(self, player: Player):
-        return self.get_army_score(self.get_player_option(player.id, 'Army')) > 0
-
-    async def on_game_end(self):
-        await super().on_game_end()
-        if self.validity != ValidityState.VALID:
-            return
-
-    @property
-    def is_draw(self):
-        for army in self.armies:
-            for result in self._results[army]:
-                if result[1] == 'draw':
-                    return True
-        return False
-
-    def get_army_score(self, army):
+        return self.outcome(player) == GameOutcome.VICTORY
+
+    def get_army_score(self, army: int) -> int:
         """
-        The head-to-head matchup ranking uses only win/loss as a factor
-        :param army:
-        :return:
+        We override this function so that ladder game scores are only reported
+        as 1 for win and 0 for anything else.
         """
-        if army not in self._results:
-            return 0
-
-        for result in self._results[army]:
+        for result in self._results.get(army, []):
             if result[1] == 'victory':
```

Comment:
No it won't because:
```python
for result in []:
    # Code here is never executed
```
If the list is empty, there's nothing to iterate over. That's why I chose empty list, so I could get rid of an additional edge case.